format/spec.md

```diff
@@ -1237,17 +1237,36 @@ Content file (data or delete) is serialized as a JSON object according to the fo
 | **`equality-ids`**       |`JSON list of int: Field ids used to determine row equality in equality delete files`|`[1]`|
 | **`sort-order-id`**      |`JSON int`|`1`|
 
-### File Scan Task Serialization
-
-File scan task is serialized as a JSON object according to the following table.
-
-| Metadata field       |JSON representation|Example|
-|--------------------------|--- |--- |
-| **`schema`**          |`JSON object`|`See above, read schemas instead`|
-| **`spec`**            |`JSON object`|`See above, read partition specs instead`|
-| **`data-file`**       |`JSON object`|`See above, read content file instead`|
-| **`delete-files`**    |`JSON list of objects`|`See above, read content file instead`|
-| **`residual-filter`** |`JSON object: residual filter expression`|`{"type":"eq","term":"id","value":1}`|
+### Task Serialization
```

Comment:
Its really not clear to me why file scan task is added into the spec here?  I don't think it is referenced any place in the main body of the specification?

Reply:
@emkornfield that is a fair question. @rdblue also had a question if this should be added to REST OpenAPI. My questions is that this is also used by Flink for state serialization. Hence it is not just a REST thing.

what do others think? @rdblue @jackye1995 @aokolnychyi

Reply:
What state is being serialized?  How does it relate to reading tables?

Does the flink use-case require standardization here or is it an implementation detail of flink?

Reply:
Flink source checkpoints pending splits of `FileScanTask`. The standardization of JSON serialization scan task is used by both REST OpenAPI and Flink (checkpoint and job manager -> task manager split assignment). I would imagine if Spark streaming checkpoint pending splits (scan task), it would probably also prefer to use JSON serialization (than Java serialization).

Reply:
IMO, unless we expect Spark to read the Rest checkpoints from Flink (I would guess this isn't the case), then this is really an implementation detail of both engines and doesn't really belong in the specification.

Reply:
I would definitely prefer it if this were _not_ part of the spec. I don't think there is any reason to require a specific JSON serialization and I was surprised to see it in the spec. It's great to have documentation on exactly what the parsers produce, but we have many parsers that are not covered by the table spec and are instead in other documents like the Puffin spec, View spec, or REST spec.

Reply:
got it. @rdblue should I just remove this section from the table spec? not sure what's the policy of removing an invalid section from spec.

Reply:
for now, I have reverted the spec change. we still need to decide if/how we can remove the existing spec section on file scan task JSON serialization.